samples/pipeline/unit-of-work/Core_6/Endpoint/Program.cs

```diff
@@ -18,16 +18,11 @@ static void Main()
         endpointConfiguration.EnableInstallers();
         endpointConfiguration.SendFailedMessagesTo("error");
 
+        var sessionProvider = new MySessionProvider();
+
         #region configuration
-        endpointConfiguration.RegisterComponents(
-            registration: components =>
-            {
-                components.ConfigureComponent(
-                    componentFactory: builder => new MySessionProvider(),
-                    dependencyLifecycle: DependencyLifecycle.SingleInstance);
-            });
         var pipeline = endpointConfiguration.Pipeline;
-        pipeline.Register(typeof(MyUowBehavior), "Manages the session");
+        pipeline.Register(_ => new MyUowBehavior(sessionProvider), "Manages the session");
```

Comment:
If I recall correctly you don't need the lambda overload there is one where you can pass in the behavior instance directly